docs/src/internals/component.md

```diff
@@ -0,0 +1,20 @@
+# Model components
+
+## @defcomp
+The macro ```@defcomp``` defines model components. Some changes are in store:
+
+1. Code simplification using the ```@capture``` macro from MacroTools.jl.
+
+1. Remove the explicit definition of indices via ```regions = Index()``` since regions can be defined on first usage. In any case, indices must be defined (with actual values) at the model level, so ensuring this is the more relevant validity check.
```

Comment:
We can merge as is, I'm a little unclear right now whether we want to really do away with it entirely, though. I could imagine something where we keep this, but one passes the default index values in, like ``regions = Index([:US, :WEU])`` or something like that.